src/pkg/packager/remove.go

```diff
@@ -98,10 +80,35 @@ func (p *Packager) Remove(packageName string) error {
 
 					return err
 				}
+
+				// Remove the uninstalled chart from the list of installed charts
+				deployedPackage.DeployedComponents[i].InstalledCharts = deployedPackage.DeployedComponents[i].InstalledCharts[:h]
+				p.updatePackageSecret(deployedPackage, secretName)
 			}
+
+			// Remove the component we just removed from the array
+			deployedPackage.DeployedComponents = append(deployedPackage.DeployedComponents[:i], deployedPackage.DeployedComponents[i+1:]...)
+		}
+
+		if len(deployedPackage.DeployedComponents) == 0 {
+			// All the installed components were deleted, there for this package is no longer actually deployed
+			_ = p.cluster.Kube.DeleteSecret(packageSecret)
+		} else {
+			p.updatePackageSecret(deployedPackage, secretName)
 		}
-		p.cluster.Kube.DeleteSecret(packageSecret)
 	}
 
 	return nil
 }
+
+func (p *Packager) updatePackageSecret(deployedPackage types.DeployedPackage, secretName string) {
+	// Save the new secret with the removed components removed from the secret
+	newPackageSecret := p.cluster.Kube.GenerateSecret("zarf", secretName, corev1.SecretTypeOpaque)
+	newPackageSecret.Labels["package-deploy-info"] = p.cfg.Pkg.Metadata.Name
```

Comment:
We should make `"package-deploy-info"` a static const now that we are doing that more.